sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/CheckpointStartingPosition.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Messaging.EventHubs.Consumer
+{
+    /// <summary>
+    ///   The position in a partition's event stream to use when updating a checkpoint, indicates that an event processor should begin reading from the next event.
+    /// </summary>
+    ///
+    public struct CheckpointStartingPosition
+    {
+        /// <summary>
+        ///   The offset to associate with the checkpoint. If there is a <see cref="SequenceNumber"/> associated with this checkpoint, then this value will be used for
+        ///   informational metadata. If no <see cref="SequenceNumber"/> is associated, then this will be used for positioning when events are read.
+        /// </summary>
+        ///
+        public long? Offset { get; }
+
+        /// <summary>
+        ///   The replication segment to associate with the checkpoint. Used in conjunction with the sequence number if using a geo replication enabled Event Hubs namespace.
+        /// </summary>
+        ///
+        public string ReplicationSegment { get; }
+
+        /// <summary>
+        ///   The sequence number to associate with the checkpoint. If populated, it indicates that a processor should begin reading from the next event in the stream.
+        /// </summary>
+        ///
+        public long? SequenceNumber { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="CheckpointStartingPosition"/> struct.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   This constructor is generally used when updating checkpoints for non-Geo-DR enabled namespaces.
+        /// </remarks>
+        ///
+        /// <param name="offset">The offset to associate with the checkpoint, indicating that a processor should begin reading from the next event in the stream.</param>
+        ///
+        public CheckpointStartingPosition(long offset)
```

Comment:
Since we're defaulting to sequence number for checkpointing, I dropped the sequence number here. The change makes the "informational sequence number" concept obsolete so it made more sense to me to isolate the two code paths:
- offset (where offset determines the positioning)
- sequence number + optional replication segment + optional informational offset (where the sequence number determines the positioning)

I didn't want to do a ctor with `(long offset, long? sequenceNumber)` because to me that means offset is determining the position. I also didn't want to do one with or starting with `(long sequenceNumber, long? offset)` because:
- say you have : `processorClient.UpdateCheckpointAsync("partition", 10, 20, ct)`
- then it seems to tempting to just switch to:  processorClient.UpdateCheckpointAsync("partition", new CheckpointStartingPosition(10, 20) ct) but that will flip offset/ sequenceNumber

Reply:
Ok got it. 👍 

Do you think my concern:

> say you have : `processorClient.UpdateCheckpointAsync("partition", 10, 20, ct)`
then it seems to tempting to just switch to: `processorClient.UpdateCheckpointAsync("partition", new CheckpointStartingPosition(10, 20) ct)` but that will flip `offset`/ `sequenceNumber`

is worth worrying about at all? If not then we can just drop that one.

Reply:
That's an interesting one that I overlooked.   That may indeed be a rough spot, but I don't know what to do about it.